source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -666,4 +738,93 @@ namespace AdaptiveCards::Rendering::Uwp
         auto foundSvg = url.find("svg");
         return !(foundSvg == std::string::npos);
     }
+
+    void render_xaml::XamlBuilder::ParseXmlForHeightAndWidth(winrt::XmlDocument const& xmlDoc,
+                                                             winrt::SvgImageSource const& imageSource)
+    {
+        if (xmlDoc)
+        {
+            auto rootElement = xmlDoc.DocumentElement();
+            auto height = rootElement.GetAttribute(L"height");
+            auto width = rootElement.GetAttribute(L"width");
+
+            // We only need to set height or width, not both (fixes aspect ratio for person style)
+            bool isHeightSet = false;
+
+            if (!height.empty())
+            {
+                if (auto heightAsDouble = TryHStringToDouble(height))
+                {
+                    SetRasterizedPixelHeightAsync(imageSource, heightAsDouble.value());
+                    isHeightSet = true;
+                }
+            }
+
+            if (!width.empty())
+            {
+                if (auto widthAsDouble = TryHStringToDouble(width))
+                {
+                    SetRasterizedPixelWidthAsync(imageSource, widthAsDouble.value(), isHeightSet);
+                }
+            }
+        }
+    }
+
+    winrt::fire_and_forget render_xaml::XamlBuilder::SetRasterizedPixelHeightAsync(winrt::SvgImageSource const imageSource,
+                                                                           double const imageSize,
+                                                                           bool const dropIfUnset)
+    {
+        co_await winrt::resume_foreground(imageSource.Dispatcher());
+        auto currentSize = imageSource.RasterizePixelHeight();
+        bool sizeIsUnset = isinf(currentSize);
+
+        bool dropHeight = sizeIsUnset && dropIfUnset;
+
+        if (!dropHeight)
+        {
+            imageSource.RasterizePixelHeight(imageSize);
+        }
+    }
+
+    winrt::fire_and_forget render_xaml::XamlBuilder::SetRasterizedPixelWidthAsync(winrt::SvgImageSource const imageSource,
+                                                                          double const imageSize,
+                                                                          bool const dropIfUnset)
+    {
+        co_await winrt::resume_foreground(imageSource.Dispatcher());
+        auto currentSize = imageSource.RasterizePixelWidth();
+        bool sizeIsUnset = isinf(currentSize);
+
+        bool dropWidth = sizeIsUnset && dropIfUnset;
```

Comment:
I don't seem to follow the logic here... how's that relevant if size is set or unset if you want to drop it if height is set anyway?

Reply:
I can add a comment here explaining this.

If an image has an explicit size, we have already set it from within `BuildImage`. If that is the case, we _must_ update the value.

Say the image is 100x100, but the explicit size is 40x40. Without replacing the value here, the rasterized values would end up being 100x40. This would look something like this:

<img width="46" alt="image" src="https://user-images.githubusercontent.com/98650930/219502832-c9f093f8-91ff-41cf-8e6b-289e2656020b.png">

Instead of:
<img width="47" alt="image" src="https://user-images.githubusercontent.com/98650930/219502873-f4cf660d-7840-413b-a179-8922ffb5862d.png">